{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from glob import glob\n",
    "import networkx as nx\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>chr</th>\n",
       "      <th>site</th>\n",
       "      <th>strand</th>\n",
       "      <th>splicesite</th>\n",
       "      <th>STARannot</th>\n",
       "      <th>dinuc</th>\n",
       "      <th>cell</th>\n",
       "      <th>rna_type</th>\n",
       "      <th>reads</th>\n",
       "      <th>gene</th>\n",
       "      <th>cryptic_status</th>\n",
       "      <th>cryptic_type</th>\n",
       "      <th>exon</th>\n",
       "      <th>constitutiveexon</th>\n",
       "      <th>intron</th>\n",
       "      <th>constitutiveintron</th>\n",
       "      <th>tpm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>100001026</td>\n",
       "      <td>-</td>\n",
       "      <td>3ss</td>\n",
       "      <td>0</td>\n",
       "      <td>GA</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>nascent</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000107554</td>\n",
       "      <td>cryptic</td>\n",
       "      <td>LowFidelity</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000107554:99972135-100009837</td>\n",
       "      <td>0</td>\n",
       "      <td>42.279118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>100001644</td>\n",
       "      <td>-</td>\n",
       "      <td>5ss</td>\n",
       "      <td>0</td>\n",
       "      <td>GC</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>nascent</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000107554</td>\n",
       "      <td>cryptic</td>\n",
       "      <td>HighFidelity</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000107554:99972135-100009837</td>\n",
       "      <td>0</td>\n",
       "      <td>42.279118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>100002941</td>\n",
       "      <td>-</td>\n",
       "      <td>5ss</td>\n",
       "      <td>0</td>\n",
       "      <td>GT</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>nascent</td>\n",
       "      <td>30</td>\n",
       "      <td>ENSG00000107554</td>\n",
       "      <td>cryptic</td>\n",
       "      <td>HighFidelity</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000107554:99972135-100009837</td>\n",
       "      <td>0</td>\n",
       "      <td>42.279118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>100003024</td>\n",
       "      <td>-</td>\n",
       "      <td>3ss</td>\n",
       "      <td>0</td>\n",
       "      <td>AG</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>nascent</td>\n",
       "      <td>30</td>\n",
       "      <td>ENSG00000107554</td>\n",
       "      <td>cryptic</td>\n",
       "      <td>HighFidelity</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000107554:99972135-100009837</td>\n",
       "      <td>0</td>\n",
       "      <td>42.279118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>100006236</td>\n",
       "      <td>-</td>\n",
       "      <td>5ss</td>\n",
       "      <td>0</td>\n",
       "      <td>GT</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>nascent</td>\n",
       "      <td>7</td>\n",
       "      <td>ENSG00000107554</td>\n",
       "      <td>cryptic</td>\n",
       "      <td>recursive</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000107554:99972135-100009837</td>\n",
       "      <td>0</td>\n",
       "      <td>42.279118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2622128</th>\n",
       "      <td>364130</td>\n",
       "      <td>X</td>\n",
       "      <td>9939367</td>\n",
       "      <td>+</td>\n",
       "      <td>5ss</td>\n",
       "      <td>1</td>\n",
       "      <td>GT</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>polya</td>\n",
       "      <td>94</td>\n",
       "      <td>ENSG00000146950</td>\n",
       "      <td>canonical</td>\n",
       "      <td>canonical</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000146950:9939367-9944640;ENSG0000014695...</td>\n",
       "      <td>0</td>\n",
       "      <td>3.752241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2622129</th>\n",
       "      <td>364131</td>\n",
       "      <td>X</td>\n",
       "      <td>9944640</td>\n",
       "      <td>+</td>\n",
       "      <td>3ss</td>\n",
       "      <td>1</td>\n",
       "      <td>AG</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>polya</td>\n",
       "      <td>75</td>\n",
       "      <td>ENSG00000146950</td>\n",
       "      <td>canonical</td>\n",
       "      <td>canonical</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000146950:9939367-9944640;ENSG0000014695...</td>\n",
       "      <td>0</td>\n",
       "      <td>3.752241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2622130</th>\n",
       "      <td>364132</td>\n",
       "      <td>X</td>\n",
       "      <td>9944643</td>\n",
       "      <td>+</td>\n",
       "      <td>3ss</td>\n",
       "      <td>1</td>\n",
       "      <td>AG</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>polya</td>\n",
       "      <td>19</td>\n",
       "      <td>ENSG00000146950</td>\n",
       "      <td>canonical</td>\n",
       "      <td>canonical</td>\n",
       "      <td>ENSG00000146950:9944641-9944913</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000146950:9939367-9944643</td>\n",
       "      <td>0</td>\n",
       "      <td>3.752241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2622131</th>\n",
       "      <td>364133</td>\n",
       "      <td>X</td>\n",
       "      <td>9944914</td>\n",
       "      <td>+</td>\n",
       "      <td>5ss</td>\n",
       "      <td>1</td>\n",
       "      <td>GT</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>polya</td>\n",
       "      <td>114</td>\n",
       "      <td>ENSG00000146950</td>\n",
       "      <td>canonical</td>\n",
       "      <td>canonical</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000146950:9944914-9946670</td>\n",
       "      <td>0</td>\n",
       "      <td>3.752241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2622132</th>\n",
       "      <td>364134</td>\n",
       "      <td>X</td>\n",
       "      <td>9946670</td>\n",
       "      <td>+</td>\n",
       "      <td>3ss</td>\n",
       "      <td>1</td>\n",
       "      <td>AG</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>polya</td>\n",
       "      <td>114</td>\n",
       "      <td>ENSG00000146950</td>\n",
       "      <td>canonical</td>\n",
       "      <td>canonical</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000146950:9944914-9946670</td>\n",
       "      <td>0</td>\n",
       "      <td>3.752241</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2622133 rows Ã— 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          index chr       site strand splicesite  STARannot dinuc   cell  \\\n",
       "0             0  10  100001026      -        3ss          0    GA  KNS60   \n",
       "1             1  10  100001644      -        5ss          0    GC  KNS60   \n",
       "2             2  10  100002941      -        5ss          0    GT  KNS60   \n",
       "3             3  10  100003024      -        3ss          0    AG  KNS60   \n",
       "4             4  10  100006236      -        5ss          0    GT  KNS60   \n",
       "...         ...  ..        ...    ...        ...        ...   ...    ...   \n",
       "2622128  364130   X    9939367      +        5ss          1    GT  KNS60   \n",
       "2622129  364131   X    9944640      +        3ss          1    AG  KNS60   \n",
       "2622130  364132   X    9944643      +        3ss          1    AG  KNS60   \n",
       "2622131  364133   X    9944914      +        5ss          1    GT  KNS60   \n",
       "2622132  364134   X    9946670      +        3ss          1    AG  KNS60   \n",
       "\n",
       "        rna_type  reads             gene cryptic_status  cryptic_type  \\\n",
       "0        nascent      5  ENSG00000107554        cryptic   LowFidelity   \n",
       "1        nascent      5  ENSG00000107554        cryptic  HighFidelity   \n",
       "2        nascent     30  ENSG00000107554        cryptic  HighFidelity   \n",
       "3        nascent     30  ENSG00000107554        cryptic  HighFidelity   \n",
       "4        nascent      7  ENSG00000107554        cryptic     recursive   \n",
       "...          ...    ...              ...            ...           ...   \n",
       "2622128    polya     94  ENSG00000146950      canonical     canonical   \n",
       "2622129    polya     75  ENSG00000146950      canonical     canonical   \n",
       "2622130    polya     19  ENSG00000146950      canonical     canonical   \n",
       "2622131    polya    114  ENSG00000146950      canonical     canonical   \n",
       "2622132    polya    114  ENSG00000146950      canonical     canonical   \n",
       "\n",
       "                                    exon constitutiveexon  \\\n",
       "0                                      0              NaN   \n",
       "1                                      0              NaN   \n",
       "2                                      0              NaN   \n",
       "3                                      0              NaN   \n",
       "4                                      0              NaN   \n",
       "...                                  ...              ...   \n",
       "2622128                                0              NaN   \n",
       "2622129                                0              NaN   \n",
       "2622130  ENSG00000146950:9944641-9944913                0   \n",
       "2622131                                0              NaN   \n",
       "2622132                                0              NaN   \n",
       "\n",
       "                                                    intron constitutiveintron  \\\n",
       "0                       ENSG00000107554:99972135-100009837                  0   \n",
       "1                       ENSG00000107554:99972135-100009837                  0   \n",
       "2                       ENSG00000107554:99972135-100009837                  0   \n",
       "3                       ENSG00000107554:99972135-100009837                  0   \n",
       "4                       ENSG00000107554:99972135-100009837                  0   \n",
       "...                                                    ...                ...   \n",
       "2622128  ENSG00000146950:9939367-9944640;ENSG0000014695...                  0   \n",
       "2622129  ENSG00000146950:9939367-9944640;ENSG0000014695...                  0   \n",
       "2622130                    ENSG00000146950:9939367-9944643                  0   \n",
       "2622131                    ENSG00000146950:9944914-9946670                  0   \n",
       "2622132                    ENSG00000146950:9944914-9946670                  0   \n",
       "\n",
       "               tpm  \n",
       "0        42.279118  \n",
       "1        42.279118  \n",
       "2        42.279118  \n",
       "3        42.279118  \n",
       "4        42.279118  \n",
       "...            ...  \n",
       "2622128   3.752241  \n",
       "2622129   3.752241  \n",
       "2622130   3.752241  \n",
       "2622131   3.752241  \n",
       "2622132   3.752241  \n",
       "\n",
       "[2622133 rows x 18 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files =  glob('../../../../../../../../Volumes/sandisk1/datasets/splicing/Athma_cryptic_splicing/splicesites_for_cshl/*.csv')\n",
    "dfs = []\n",
    "for f in files:\n",
    "    dfs.append(pd.read_csv(f,dtype={\"chr\": str, \"site\": int, 'strand':str, 'splicesite':str,'STARannot':int,'dinuc':str,'cell':str,\n",
    "                                    'rna_type':str,\n",
    "                                    'reads':int,\n",
    "                                    'gene':str,\n",
    "                                    'cryptic_status':str,\n",
    "                                    'cryptic_type':str,\n",
    "                                    'exon':str,\n",
    "                                    'constitutiveexon':str,\n",
    "                                    'intron':str,\n",
    "                                    'constitutiveintron':str,\n",
    "                                    'tpm':float}))\n",
    "dfs = pd.concat(dfs).reset_index()\n",
    "dfs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/5_/tjkj20xs1zjbm1r6n07ccp4h0000gn/T/ipykernel_56013/2407871938.py:4: FutureWarning: The previous implementation of stack is deprecated and will be removed in a future version of pandas. See the What's New notes for pandas 2.1.0 for details. Specify future_stack=True to adopt the new implementation and silence this warning.\n",
      "  full_tpm_df = full_tpm_df.stack(['cell', 'rna_type']).rename('tpm').to_frame().sort_index()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>tpm</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genes</th>\n",
       "      <th>cell</th>\n",
       "      <th>rna_type</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">ENSG00000000003</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">K562</th>\n",
       "      <th>cycloheximide</th>\n",
       "      <td>0.054959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nascent</th>\n",
       "      <td>0.007408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nuclear</th>\n",
       "      <td>0.110831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>polya</th>\n",
       "      <td>0.038089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KNS60</th>\n",
       "      <th>cycloheximide</th>\n",
       "      <td>10.946019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">ENSG00000283696</th>\n",
       "      <th>K562</th>\n",
       "      <th>polya</th>\n",
       "      <td>1.582833</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">KNS60</th>\n",
       "      <th>cycloheximide</th>\n",
       "      <td>0.616325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nascent</th>\n",
       "      <td>4.077657</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nuclear</th>\n",
       "      <td>1.198403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>polya</th>\n",
       "      <td>1.140936</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>120008 rows Ã— 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           tpm\n",
       "genes           cell  rna_type                \n",
       "ENSG00000000003 K562  cycloheximide   0.054959\n",
       "                      nascent         0.007408\n",
       "                      nuclear         0.110831\n",
       "                      polya           0.038089\n",
       "                KNS60 cycloheximide  10.946019\n",
       "...                                        ...\n",
       "ENSG00000283696 K562  polya           1.582833\n",
       "                KNS60 cycloheximide   0.616325\n",
       "                      nascent         4.077657\n",
       "                      nuclear         1.198403\n",
       "                      polya           1.140936\n",
       "\n",
       "[120008 rows x 1 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_tpm_df = pd.read_csv('../../../../../../../../Volumes/sandisk1/datasets/splicing/Athma_cryptic_splicing/tpms_K562_KNS60_112924.csv', index_col=0)\n",
    "full_tpm_df.columns = pd.MultiIndex.from_tuples([(r1, r2) for r1, r2 in full_tpm_df.columns.str.split('_')], names=['cell','rna_type'])\n",
    "full_tpm_df = full_tpm_df.loc[dfs.gene.unique()]\n",
    "full_tpm_df = full_tpm_df.stack(['cell', 'rna_type']).rename('tpm').to_frame().sort_index()\n",
    "\n",
    "full_tpm_df.to_csv('../../../../../../../../Volumes/sandisk1/datasets/splicing/Athma_cryptic_splicing/processed_round1/tpms.csv')\n",
    "full_tpm_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>splicesite</th>\n",
       "      <th>cell</th>\n",
       "      <th>rna_type</th>\n",
       "      <th>dinuc</th>\n",
       "      <th>cryptic_type</th>\n",
       "      <th>reads</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th>chr</th>\n",
       "      <th>strand</th>\n",
       "      <th>site</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">ENSG00000000003</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">X</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">-</th>\n",
       "      <th>100629987</th>\n",
       "      <td>3ss</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>nascent</td>\n",
       "      <td>AG</td>\n",
       "      <td>canonical</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100629987</th>\n",
       "      <td>3ss</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>cycloheximide</td>\n",
       "      <td>AG</td>\n",
       "      <td>canonical</td>\n",
       "      <td>302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100629987</th>\n",
       "      <td>3ss</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>nuclear</td>\n",
       "      <td>AG</td>\n",
       "      <td>canonical</td>\n",
       "      <td>691</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100629987</th>\n",
       "      <td>3ss</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>polya</td>\n",
       "      <td>AG</td>\n",
       "      <td>canonical</td>\n",
       "      <td>979</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100630758</th>\n",
       "      <td>5ss</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>nascent</td>\n",
       "      <td>GT</td>\n",
       "      <td>canonical</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">ENSG00000283689</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">16</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">+</th>\n",
       "      <th>54288560</th>\n",
       "      <td>3ss</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>cycloheximide</td>\n",
       "      <td>AG</td>\n",
       "      <td>canonical</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54288560</th>\n",
       "      <td>3ss</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>nuclear</td>\n",
       "      <td>AG</td>\n",
       "      <td>canonical</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54288560</th>\n",
       "      <td>3ss</td>\n",
       "      <td>KNS60</td>\n",
       "      <td>polya</td>\n",
       "      <td>AG</td>\n",
       "      <td>canonical</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">ENSG00000283696</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">1</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">+</th>\n",
       "      <th>161399913</th>\n",
       "      <td>5ss</td>\n",
       "      <td>K562</td>\n",
       "      <td>nascent</td>\n",
       "      <td>GT</td>\n",
       "      <td>canonical</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161421931</th>\n",
       "      <td>3ss</td>\n",
       "      <td>K562</td>\n",
       "      <td>nascent</td>\n",
       "      <td>AG</td>\n",
       "      <td>canonical</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2622133 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     splicesite   cell       rna_type dinuc  \\\n",
       "gene            chr strand site                                               \n",
       "ENSG00000000003 X   -      100629987        3ss  KNS60        nascent    AG   \n",
       "                           100629987        3ss  KNS60  cycloheximide    AG   \n",
       "                           100629987        3ss  KNS60        nuclear    AG   \n",
       "                           100629987        3ss  KNS60          polya    AG   \n",
       "                           100630758        5ss  KNS60        nascent    GT   \n",
       "...                                         ...    ...            ...   ...   \n",
       "ENSG00000283689 16  +      54288560         3ss  KNS60  cycloheximide    AG   \n",
       "                           54288560         3ss  KNS60        nuclear    AG   \n",
       "                           54288560         3ss  KNS60          polya    AG   \n",
       "ENSG00000283696 1   +      161399913        5ss   K562        nascent    GT   \n",
       "                           161421931        3ss   K562        nascent    AG   \n",
       "\n",
       "                                     cryptic_type  reads  \n",
       "gene            chr strand site                           \n",
       "ENSG00000000003 X   -      100629987    canonical     41  \n",
       "                           100629987    canonical    302  \n",
       "                           100629987    canonical    691  \n",
       "                           100629987    canonical    979  \n",
       "                           100630758    canonical     41  \n",
       "...                                           ...    ...  \n",
       "ENSG00000283689 16  +      54288560     canonical     64  \n",
       "                           54288560     canonical     30  \n",
       "                           54288560     canonical    108  \n",
       "ENSG00000283696 1   +      161399913    canonical     15  \n",
       "                           161421931    canonical     15  \n",
       "\n",
       "[2622133 rows x 6 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "junction_df = dfs.set_index(['gene', 'chr', 'strand', 'site'])[['splicesite','cell', 'rna_type', 'dinuc','cryptic_type','reads']].sort_index()\n",
    "junction_df.to_csv('../../../../../../../../Volumes/sandisk1/datasets/splicing/Athma_cryptic_splicing/processed_round1/junctions.csv')\n",
    "junction_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Gene Internal ID</th>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th>Gene Synbol</th>\n",
       "      <th>HasParalog</th>\n",
       "      <th>Paralog stable ID</th>\n",
       "      <th>Paralog Synbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>319832</td>\n",
       "      <td>ENSG00000284532</td>\n",
       "      <td>MIR4723</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>319833</td>\n",
       "      <td>ENSG00000238933</td>\n",
       "      <td>RF00019</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000202361</td>\n",
       "      <td>RF00019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>319833</td>\n",
       "      <td>ENSG00000238933</td>\n",
       "      <td>RF00019</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000201843</td>\n",
       "      <td>RF00019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>319833</td>\n",
       "      <td>ENSG00000238933</td>\n",
       "      <td>RF00019</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000200008</td>\n",
       "      <td>RF00019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>319833</td>\n",
       "      <td>ENSG00000238933</td>\n",
       "      <td>RF00019</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000201466</td>\n",
       "      <td>RF00019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3618911</th>\n",
       "      <td>3618911</td>\n",
       "      <td>324201</td>\n",
       "      <td>ENSG00000285013</td>\n",
       "      <td>AC243837.3</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3618912</th>\n",
       "      <td>3618912</td>\n",
       "      <td>324202</td>\n",
       "      <td>ENSG00000285154</td>\n",
       "      <td>AC243837.4</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3618913</th>\n",
       "      <td>3618913</td>\n",
       "      <td>324203</td>\n",
       "      <td>ENSG00000284933</td>\n",
       "      <td>ARHGEF5</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3618914</th>\n",
       "      <td>3618914</td>\n",
       "      <td>324204</td>\n",
       "      <td>ENSG00000285363</td>\n",
       "      <td>MTRF1LP2</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3618915</th>\n",
       "      <td>3618915</td>\n",
       "      <td>324205</td>\n",
       "      <td>ENSG00000285114</td>\n",
       "      <td>GSDMC</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3618916 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Unnamed: 0  Gene Internal ID   Gene stable ID Gene Synbol  \\\n",
       "0                 0            319832  ENSG00000284532     MIR4723   \n",
       "1                 1            319833  ENSG00000238933     RF00019   \n",
       "2                 2            319833  ENSG00000238933     RF00019   \n",
       "3                 3            319833  ENSG00000238933     RF00019   \n",
       "4                 4            319833  ENSG00000238933     RF00019   \n",
       "...             ...               ...              ...         ...   \n",
       "3618911     3618911            324201  ENSG00000285013  AC243837.3   \n",
       "3618912     3618912            324202  ENSG00000285154  AC243837.4   \n",
       "3618913     3618913            324203  ENSG00000284933     ARHGEF5   \n",
       "3618914     3618914            324204  ENSG00000285363    MTRF1LP2   \n",
       "3618915     3618915            324205  ENSG00000285114       GSDMC   \n",
       "\n",
       "         HasParalog Paralog stable ID Paralog Synbol  \n",
       "0             False               NaN            NaN  \n",
       "1              True   ENSG00000202361        RF00019  \n",
       "2              True   ENSG00000201843        RF00019  \n",
       "3              True   ENSG00000200008        RF00019  \n",
       "4              True   ENSG00000201466        RF00019  \n",
       "...             ...               ...            ...  \n",
       "3618911       False               NaN            NaN  \n",
       "3618912       False               NaN            NaN  \n",
       "3618913       False               NaN            NaN  \n",
       "3618914       False               NaN            NaN  \n",
       "3618915       False               NaN            NaN  \n",
       "\n",
       "[3618916 rows x 7 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "para = pd.read_csv('../../../../../../../../Volumes/sandisk1/datasets/splicing/Athma_cryptic_splicing/GRCh38_ensembl95.Paralogs.csv')\n",
    "para\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome/scaffold name</th>\n",
       "      <th>Gene start (bp)</th>\n",
       "      <th>Gene end (bp)</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Gene name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000284532</th>\n",
       "      <td>17</td>\n",
       "      <td>28360654</td>\n",
       "      <td>28360734</td>\n",
       "      <td>1</td>\n",
       "      <td>MIR4723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000238933</th>\n",
       "      <td>9</td>\n",
       "      <td>61642383</td>\n",
       "      <td>61642494</td>\n",
       "      <td>1</td>\n",
       "      <td>RF00019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000275693</th>\n",
       "      <td>10</td>\n",
       "      <td>88992370</td>\n",
       "      <td>88992539</td>\n",
       "      <td>1</td>\n",
       "      <td>RF02116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000275451</th>\n",
       "      <td>15</td>\n",
       "      <td>62343029</td>\n",
       "      <td>62343138</td>\n",
       "      <td>1</td>\n",
       "      <td>MIR6085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000222870</th>\n",
       "      <td>7</td>\n",
       "      <td>94495299</td>\n",
       "      <td>94495404</td>\n",
       "      <td>1</td>\n",
       "      <td>RNU6-1328P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000285013</th>\n",
       "      <td>CHR_HSCHR7_3_CTG4_4</td>\n",
       "      <td>144396023</td>\n",
       "      <td>144396316</td>\n",
       "      <td>1</td>\n",
       "      <td>AC243837.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000285154</th>\n",
       "      <td>CHR_HSCHR7_3_CTG4_4</td>\n",
       "      <td>144414782</td>\n",
       "      <td>144415702</td>\n",
       "      <td>1</td>\n",
       "      <td>AC243837.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000284933</th>\n",
       "      <td>CHR_HSCHR7_3_CTG4_4</td>\n",
       "      <td>144457568</td>\n",
       "      <td>144474951</td>\n",
       "      <td>1</td>\n",
       "      <td>ARHGEF5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000285363</th>\n",
       "      <td>CHR_HSCHR8_7_CTG7</td>\n",
       "      <td>129728744</td>\n",
       "      <td>129730445</td>\n",
       "      <td>1</td>\n",
       "      <td>MTRF1LP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000285114</th>\n",
       "      <td>CHR_HSCHR8_7_CTG7</td>\n",
       "      <td>129757225</td>\n",
       "      <td>129795938</td>\n",
       "      <td>-1</td>\n",
       "      <td>GSDMC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>64914 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Chromosome/scaffold name  Gene start (bp)  Gene end (bp)  \\\n",
       "Gene stable ID                                                             \n",
       "ENSG00000284532                       17         28360654       28360734   \n",
       "ENSG00000238933                        9         61642383       61642494   \n",
       "ENSG00000275693                       10         88992370       88992539   \n",
       "ENSG00000275451                       15         62343029       62343138   \n",
       "ENSG00000222870                        7         94495299       94495404   \n",
       "...                                  ...              ...            ...   \n",
       "ENSG00000285013      CHR_HSCHR7_3_CTG4_4        144396023      144396316   \n",
       "ENSG00000285154      CHR_HSCHR7_3_CTG4_4        144414782      144415702   \n",
       "ENSG00000284933      CHR_HSCHR7_3_CTG4_4        144457568      144474951   \n",
       "ENSG00000285363        CHR_HSCHR8_7_CTG7        129728744      129730445   \n",
       "ENSG00000285114        CHR_HSCHR8_7_CTG7        129757225      129795938   \n",
       "\n",
       "                 Strand   Gene name  \n",
       "Gene stable ID                       \n",
       "ENSG00000284532       1     MIR4723  \n",
       "ENSG00000238933       1     RF00019  \n",
       "ENSG00000275693       1     RF02116  \n",
       "ENSG00000275451       1     MIR6085  \n",
       "ENSG00000222870       1  RNU6-1328P  \n",
       "...                 ...         ...  \n",
       "ENSG00000285013       1  AC243837.3  \n",
       "ENSG00000285154       1  AC243837.4  \n",
       "ENSG00000284933       1     ARHGEF5  \n",
       "ENSG00000285363       1    MTRF1LP2  \n",
       "ENSG00000285114      -1       GSDMC  \n",
       "\n",
       "[64914 rows x 5 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_data = pd.read_csv('../../../../../../../../Volumes/sandisk1/datasets/splicing/Athma_cryptic_splicing/GRCh38_ensembl95.genes.csv')\n",
    "gene_data = gene_data.rename(columns={'Gene Chromosome':'Chromosome/scaffold name', 'Gene Strand':'Strand','Gene Synbol':'Gene name'})\n",
    "gene_data = gene_data[['Gene stable ID','Chromosome/scaffold name','Gene start (bp)','Gene end (bp)','Strand','Gene name']].set_index('Gene stable ID')\n",
    "print(gene_data.index.has_duplicates)\n",
    "gene_data.to_csv('../../../../../../../../Volumes/sandisk1/datasets/splicing/Athma_cryptic_splicing/processed_round1/gene_data.csv')\n",
    "gene_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "219472     (ENSG00000198886, ENSG00000198786)\n",
       "219575     (ENSG00000198786, ENSG00000198886)\n",
       "356712     (ENSG00000142552, ENSG00000128595)\n",
       "356713     (ENSG00000142552, ENSG00000117906)\n",
       "356714     (ENSG00000142552, ENSG00000078808)\n",
       "                          ...                \n",
       "2390973    (ENSG00000120451, ENSG00000109762)\n",
       "2390974    (ENSG00000120451, ENSG00000135317)\n",
       "2391089    (ENSG00000187735, ENSG00000204219)\n",
       "2391090    (ENSG00000187735, ENSG00000171703)\n",
       "2391091    (ENSG00000187735, ENSG00000176896)\n",
       "Length: 90528, dtype: object"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "only_observed_orthos = para[para['Gene stable ID'].isin(dfs['gene'].unique())&para['Paralog stable ID'].isin(dfs['gene'].unique())]\n",
    "only_observed_orthos = only_observed_orthos[['Gene stable ID', 'Paralog stable ID']]\n",
    "\n",
    "only_observed_orthos = only_observed_orthos.apply(lambda x: (x['Gene stable ID'],x['Paralog stable ID']),\n",
    "                                                            axis=1)\n",
    "only_observed_orthos"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G.number_of_nodes()=15001\n",
      "These nodes are missing from the paralog search\n",
      "Index([], dtype='object')\n",
      "captured 8076 ortho groups\n",
      "[0]\n",
      "1767\n"
     ]
    }
   ],
   "source": [
    "G =  nx.Graph()\n",
    "G.add_nodes_from(dfs.groupby('gene').first().apply(lambda x: (x.name, {'chr':x['chr'], 'strand':x['strand']}), axis=1))\n",
    "G.add_edges_from(only_observed_orthos)\n",
    "print(f'{G.number_of_nodes()=}')\n",
    "\n",
    "all_nodes = pd.Series(G.nodes).index\n",
    "print('These nodes are missing from the paralog search')\n",
    "print(all_nodes[~all_nodes.isin(para['Gene stable ID'].unique())])\n",
    "\n",
    "ortho_groups = list(nx.connected_components(G))\n",
    "print(f'captured {len(ortho_groups)} ortho groups')\n",
    "\n",
    "test_set = [\n",
    "    (G.degree[node], node)\n",
    "    for node, data in G.nodes(data=True) \n",
    "    if data.get('chr') in ['1', '3', '5', '7', '9'] and G.degree[node] == 0\n",
    "]\n",
    "\n",
    "neghbors, test_nodes = zip(*test_set)\n",
    "print(np.unique(neghbors))\n",
    "print(len(test_nodes))\n",
    "\n",
    "G.remove_nodes_from(test_nodes)\n",
    "\n",
    "\n",
    "ortho_groups = list(nx.connected_components(G))\n",
    "len(ortho_groups)\n",
    "\n",
    "rng = np.random.default_rng(42)\n",
    "\n",
    "perm = rng.permutation(len(ortho_groups))\n",
    "chunks = np.array_split(perm, 5)\n",
    "cvs = [[ortho_groups[i] for i in chunk] for chunk in chunks ]\n",
    "cvs = [set.union(*cv) for cv in cvs]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>Test_set</th>\n",
       "      <th>CV0</th>\n",
       "      <th>CV1</th>\n",
       "      <th>CV2</th>\n",
       "      <th>CV3</th>\n",
       "      <th>CV4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000091073</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000120616</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000158796</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000233006</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000175040</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14996</th>\n",
       "      <td>ENSG00000132912</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14997</th>\n",
       "      <td>ENSG00000025039</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14998</th>\n",
       "      <td>ENSG00000259768</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14999</th>\n",
       "      <td>ENSG00000114098</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15000</th>\n",
       "      <td>ENSG00000163874</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15001 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  gene  Test_set    CV0    CV1    CV2    CV3    CV4\n",
       "0      ENSG00000091073     False  False  False  False   True  False\n",
       "1      ENSG00000120616     False   True  False  False  False  False\n",
       "2      ENSG00000158796     False  False   True  False  False  False\n",
       "3      ENSG00000233006      True  False  False  False  False  False\n",
       "4      ENSG00000175040     False  False   True  False  False  False\n",
       "...                ...       ...    ...    ...    ...    ...    ...\n",
       "14996  ENSG00000132912      True  False  False  False  False  False\n",
       "14997  ENSG00000025039     False  False  False  False   True  False\n",
       "14998  ENSG00000259768     False  False  False  False   True  False\n",
       "14999  ENSG00000114098      True  False  False  False  False  False\n",
       "15000  ENSG00000163874     False  False  False  False   True  False\n",
       "\n",
       "[15001 rows x 7 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_to_split = pd.DataFrame()\n",
    "gene_to_split['gene'] = list(set.union(set(test_nodes), set.union(*cvs)))\n",
    "\n",
    "gene_to_split['Test_set'] = False\n",
    "gene_to_split.loc[gene_to_split['gene'].isin(test_nodes),'Test_set'] = True\n",
    "\n",
    "for i, cv in enumerate(cvs):\n",
    "    gene_to_split[f'CV{i}'] = False\n",
    "    gene_to_split.loc[gene_to_split['gene'].isin(cv),f'CV{i}'] = True\n",
    "\n",
    "gene_to_split.to_csv('../../../../../../../../Volumes/sandisk1/datasets/splicing/Athma_cryptic_splicing/processed_round1/gene_split.csv')\n",
    "gene_to_split"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Test_set    1767\n",
       "CV0         2502\n",
       "CV1         2756\n",
       "CV2         2758\n",
       "CV3         2518\n",
       "CV4         2700\n",
       "dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_to_split.drop(columns='gene').sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "MPSA_pred",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
